libarchive/archive_read_open_filename.c

```diff
@@ -372,6 +370,10 @@ file_open(struct archive *a, void *client_data)
 		mine->use_lseek = 1;
 
 	return (ARCHIVE_OK);
+done_error:
+	if (0 < fd)
```

Comment:
It might make the code clearer to be explicit about both options here:
```
   if (fd == -1 || fd == 0) { /* Don't close a nonexistent fd, don't close stdin. */
   }
```